src/query/expression/src/types/number.rs

```diff
@@ -549,24 +549,29 @@ impl<T: Number> SimpleDomain<T> {
     }
 
     pub fn overflow_cast_with_minmax<U: Number>(&self, min: U, max: U) -> (SimpleDomain<U>, bool) {
-        let (min, min_overflowing) = overflow_cast_with_minmax::<T, U>(self.min, min, max);
-        let (max, max_overflowing) = overflow_cast_with_minmax::<T, U>(self.max, min, max);
+        let (min, min_overflowing) = overflow_cast_with_minmax::<T, U>(self.min, min, max, min);
+        let (max, max_overflowing) = overflow_cast_with_minmax::<T, U>(self.max, min, max, max);
```

Comment:
fallback value is the best use case for Option, and we can use it like this:
```suggestion
        let (min, min_overflowing) = overflow_cast_with_minmax::<T, U>(self.min, min, max).unwrap_or(min);
        let (max, max_overflowing) = overflow_cast_with_minmax::<T, U>(self.max, min, max).unwrap_or(max);
```